src/hooks/init/check-update.ts

```diff
@@ -1,69 +1,135 @@
+  stat: await stat(file),
+})
+
+function hasNotBeenMsSinceDate(ms: number, date: Date): boolean {
+  const now = new Date()
+  const diff = now.getTime() - date.getTime()
+  return diff < ms
+}
+
+function convertToMs(frequency: number, unit: 'days' | 'hours' | 'milliseconds' | 'minutes' | 'seconds'): number {
+  switch (unit) {
+    case 'days': {
+      return frequency * 24 * 60 * 60 * 1000
+    }
+
+    case 'hours': {
+      return frequency * 60 * 60 * 1000
+    }
+
+    case 'milliseconds': {
+      return frequency
+    }
+
+    case 'minutes': {
+      return frequency * 60 * 1000
+    }
+
+    case 'seconds': {
+      return frequency * 1000
+    }
+
+    default: {
+      // default to minutes
+      return frequency * 60 * 1000
+    }
+  }
+}
 
 const hook: Hook<'init'> = async function ({config}) {
+  const debug = makeDebug('update-check')
   const file = join(config.cacheDir, 'version')
 
   // Destructure package.json configuration with defaults
   const {
-    timeoutInDays = 60,
+    authorization = '',
+    frequency,
+    frequencyUnit,
     message = '<%= config.name %> update available from <%= chalk.greenBright(config.version) %> to <%= chalk.greenBright(latest) %>.',
     registry = 'https://registry.npmjs.org',
-    authorization = '',
-  } = (config.pjson.oclif as any)['warn-if-update-available'] || {}
+    timeoutInDays = 60,
+  } = config.pjson.oclif['warn-if-update-available'] || {}
+
+  const warningFrequency = (config.scopedEnvVar('NEW_VERSION_CHECK_FREQ') as number | undefined) ?? frequency
+  const warningFrequencyUnit =
+    (config.scopedEnvVar('NEW_VERSION_CHECK_FREQ_UNIT') as typeof frequencyUnit) ?? frequencyUnit ?? 'minutes'
+
+  const tag = (config.scopedEnvVar('NEW_VERSION_CHECK_TAG') as string | undefined) ?? 'latest'
 
   const checkVersion = async () => {
     try {
       // do not show warning if updating
-      if (process.argv[2] === 'update') return
-      const distTags = JSON.parse(await readFile(file, 'utf8'))
+      if (process.argv[2] === 'update' || this.config.scopedEnvVarTrue('SKIP_NEW_VERSION_CHECK')) return
+
+      const {contents: distTags, stat} = await readJSON<{[tag: string]: string}>(file)
+
+      // If the file was modified before the timeout, don't show the warning
+      if (
+        warningFrequency &&
+        warningFrequencyUnit &&
+        hasNotBeenMsSinceDate(convertToMs(warningFrequency, warningFrequencyUnit), stat.mtime)
+      )
+        return
+
       if (config.version.includes('-')) {
-        // to-do: handle channels
+        // to-do: handle prerelease channels
         return
       }
-      const semverGt = await import('semver/functions/gt')
-      if (distTags && distTags.latest && semverGt(distTags.latest.split('-')[0], config.version.split('-')[0])) {
-        const chalk: Chalk = require('chalk')
+
+      const {gt} = await import('semver')
+      if (distTags[tag] && gt(distTags[tag].split('-')[0], config.version.split('-')[0])) {
```

Comment:
nit: since you've already got distTags, you could check that and avoid the semver import if it weren't needed